### ISSUE TITLE
AVM Fritz!Box 7430 bad performance

### ISSUE THREAD

**Issue by Snowiiii** [openwrt/openwrt#14545]
### Describe the bug

Hello. 2 Days ago i switched my 7430 from FritzOS to OpenWRT.
I expected same or better Internet Performance but this was not the case i got much worse performance.

Currently i pay for an 50Mibit/s download and 10Mbit/s upload speed at Vodafone. And thats what i also got with ethernet on FritzOS, but now the Results are just terrible 
![image](https://github.com/openwrt/openwrt/assets/71594357/368ebe26-a788-4043-b3b8-7edd8fb9715f)
I looked at the Internet for an solution, some people said that SQM should help but it hasn't.

These are my current wan settings:
![image](https://github.com/openwrt/openwrt/assets/71594357/584bd3dd-94b8-498a-9533-899e61786de2)
As default the device was just dsl0, This broke everything then with dsl0.7 i had an Internet connection.

So people said the it maybe an Code issue and not an Config issue so im here.

@Djfe 

### OpenWrt version

r23630-842932a63d

### OpenWrt release

23.05.2

### OpenWrt target/subtarget

lantiq/xrx200

### Device

AVM FRITZ!Box 7430

### Image kind

Official downloaded image

### Steps to reproduce

See that you have an bad internet connection

### Actual behaviour

I have an bad Internet connection and an high ping
(about 15Mbit/s download and ~1Mbit/s upload)

### Expected behaviour

To have an good Internet connection like with FritzOS
(50Mbit/s download and 10Mbit/s upload)

### Additional info

The results are from an ethernet connection, but WIFI is slow too

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by brada4** [comment]
Try this https://github.com/openwrt/openwrt/issues/13379#issuecomment-1911833638

In principle duplicate of (and full of maybe-workarounds) https://github.com/openwrt/openwrt/issues/10224

**Comment by Snowiiii** [comment]
> Try this [#13379 (comment)](https://github.com/openwrt/openwrt/issues/13379#issuecomment-1911833638)
> 
> In principle duplicate of (and full of maybe-workarounds) #10224

gives an error
![image](https://github.com/openwrt/openwrt/assets/71594357/735d45f4-2dbf-4924-811d-3c4f47bc666a)

**Comment by brada4** [comment]
You need to switch route to ip address of pppoe server from pppoo-wan to dsl0.7
Either way the problem in 10224 is still not understood well enough to fix.

**Comment by Snowiiii** [comment]
> Either way the problem in 10224 is still not understood well enough to fix.

This need to change

**Comment by brada4** [comment]
Any success with swapping interface of route to ppp server? 
i.e
```
ip -4 ro sh
<try to identify one individual route attached to pppoe-wan>
ip ro del "the full text"
ip ro add "swap pppoe-wan to dsl0.7"
service firewall restart ; sleep 61 ; service firewall restart
```

**Comment by brada4** [comment]
Also `ping -I pppoo-wan ppoe-server-ip` vs `-I dsl0.7` may show some latency weirdness.

**Comment by Snowiiii** [comment]
i dont understand

**Comment by brada4** [comment]
There is a specific route to your provider's PPPoE server.
It is (mistakenly|uncommonly) assigned via pppoe tunnel, not via physical interface. 
```
# ip -4 ro sh
default via 192.0.2.1 dev pppoe-wan  src 192.0.2.2
10.11.12.13  dev pppoe-wan  scope link # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

You need to replace device in that specific route. Then reload firewall twice with pause to assure any offload states (can be) offloaded again.

**Comment by Snowiiii** [comment]
> There is a specific route to your provider's PPPoE server. It is (mistakenly|uncommonly) assigned via pppoe tunnel, not via physical interface.
> 
> ```
> # ip -4 ro sh
> default via 192.0.2.1 dev pppoe-wan  src 192.0.2.2
> 10.11.12.13  dev pppoe-wan  scope link # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
> ```
> 
> You need to replace device in that specific route. Then reload firewall twice with pause to assure any offload states (can be) offloaded again.

looks like this
![image](https://github.com/openwrt/openwrt/assets/71594357/0deb2f98-3ebb-49bf-baa1-65959f3b50cd)
what now?

**Comment by brada4** [comment]
The second route needs to be replaced:
```
ip route del "84.59.211.1 dev pppoe-wan scope link src 2.204.253"
ip route add "84.59.211.1 dev dsl0.7 scope link"
```

**Comment by Snowiiii** [comment]
replaced with what ?, now it says: RTNETLINK anseers: No such device

**Comment by brada4** [comment]
so check DSL device name in 'ip addr'

**Comment by Snowiiii** [comment]
``` 
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1508 qdisc fq_codel state UNKNOWN qlen 1000
    link/ether 96:14:f8:bb:16:d2 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::9414:f8ff:febb:16d2/64 scope link 
       valid_lft forever preferred_lft forever
3: lan3@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 7c:ff:4d:49:c2:c0 brd ff:ff:ff:ff:ff:ff
4: lan4@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 7c:ff:4d:49:c2:c0 brd ff:ff:ff:ff:ff:ff
5: lan1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether 7c:ff:4d:49:c2:c0 brd ff:ff:ff:ff:ff:ff
6: lan2@eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether 7c:ff:4d:49:c2:c0 brd ff:ff:ff:ff:ff:ff
10: dsl0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP qlen 1000
    link/ether 7c:ff:4d:49:c2:c3 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::7eff:4dff:fe49:c2c3/64 scope link 
       valid_lft forever preferred_lft forever
29: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 7c:ff:4d:49:c2:c0 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd75:95dc:4e62::1/60 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::7eff:4dff:fe49:c2c0/64 scope link 
       valid_lft forever preferred_lft forever
30: dsl0.7@dsl0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 7c:ff:4d:49:c2:c3 brd ff:ff:ff:ff:ff:ff
31: pppoe-wan: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc fq_codel state UNKNOWN qlen 3
    link/ppp 
    inet 2.204.169.253 peer 84.59.211.1/32 scope global pppoe-wan
       valid_lft forever preferred_lft forever
32: phy0-ap0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether 00:02:03:04:05:06 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::202:3ff:fe04:506/64 scope link 
       valid_lft forever preferred_lft forever
```

**Comment by brada4** [comment]
Maybe without tickmarks indeed.

**Comment by Snowiiii** [comment]
I will test this tomorrow, my mom is a mad that the Internet stops working haha

**Comment by Snowiiii** [comment]
> Maybe without tickmarks indeed.
> 
> ```
> ip route del 84.59.211.1 dev pppoe-wan scope link src 2.204.253
> ip route add 84.59.211.1 dev dsl0.7 scope link src 2.204.253
> ```
> 
> EDIT: since no IP on the interface it needs SRC IP specified in route.

what exactly this does ?, or why it should help

**Comment by brada4** [comment]
It changes route to pppoe concentrator to physical interrface in a manner similar to debian pppoeconf

**Comment by Snowiiii** [comment]
Now both commands are failing:

![image](https://github.com/openwrt/openwrt/assets/71594357/dfd0ccbb-42d3-4845-9c9f-52b277354377)

**Comment by Snowiiii** [comment]
> The second route needs to be replaced:
> 
> ```
> ip route del "84.59.211.1 dev pppoe-wan scope link src 2.204.253"
> ip route add "84.59.211.1 dev dsl0.7 scope link"
> ```

so the first command worked here, the secound not

**Comment by brada4** [comment]
You can restore first route.
Can work without src ip, hopefully takes one from pppoewan

**Comment by Snowiiii** [comment]
> You can restore first route. Can work without src ip, hopefully takes one from pppoewan

i still don't understand, You must consider that i have openwrt since a few days and not much fimiliar with Networking at all

**Comment by Snowiiii** [comment]
I saw that for some reason dsl0.7 is disabled while dsl0 is enabled, Does it make a diffrence or is it right ?
![image](https://github.com/openwrt/openwrt/assets/71594357/f5a1af8d-c271-4eec-830d-3bcf83061ef6)

**Comment by brada4** [comment]
It is likely brought UP along pppoe interface.
but still the route spec looks invalid.

**Comment by Snowiiii** [comment]
> It is likely brought UP along pppoe interface. but still the route spec looks invalid.

i enabled dsl0.7 instead but nothing changed
![image](https://github.com/openwrt/openwrt/assets/71594357/f3180ee9-8eec-4d87-bb24-a1cddcbf6434)

**Comment by Snowiiii** [comment]
Hey i think i found the issue, I looked at https://forum.openwrt.org/t/fritz-7412-slow-internet-connection/108306/9
and they said its an CPU issue and i guess it is
![image](https://github.com/openwrt/openwrt/assets/71594357/a9ccd752-e038-429a-a8c4-05ebf5ef111b)

**Comment by brada4** [comment]
`timeout 5 ubus monitor | tee /tmp/ubus.log`
Try to figure out what is being done in tight loop
Serms dhcp requests are sent to pppoe-wan at max speed.
since you havre ip from ppp it is safe to kill dhcp client and see if cpu usage drops(or in general disable dhcp on ppp wan connection, it gets ip from ppp)

**Comment by brada4** [comment]
timeout == coreutils-timeout
just to kill outputting process even if shell is slowbto follow

**Comment by Snowiiii** [comment]
`timeout: not found`

> `timeout 5 ubus monitor | tee /tmp/ubus.log` Try to figure out what is being done in tight loop Serms dhcp requests are sent to pppoe-wan at max speed. since you havre ip from ppp it is safe to kill dhcp client and see if cpu usage drops(or in general disable dhcp on ppp wan connection, it gets ip from ppp) PS impressed by your no-knowledge of network.

**Comment by Snowiiii** [comment]
> `timeout 5 ubus monitor | tee /tmp/ubus.log` Try to figure out what is being done in tight loop Serms dhcp requests are sent to pppoe-wan at max speed. since you havre ip from ppp it is safe to kill dhcp client and see if cpu usage drops(or in general disable dhcp on ppp wan connection, it gets ip from ppp) PS impressed by your no-knowledge of network.

gives:
```
root@Baumhaus:~# timeout 5 ubus monitor | tee /tmp/ubus.log
-> 48300086 #00000003         status: {"status":0}
Terminated
``

**Comment by danpawlik** [comment]
Just curious: did you try to make factory reset and restore config from a backup if there would be still an issue? I know that's not a fix or solution...

**Comment by Snowiiii** [comment]
> Just curious: did you try to make factory reset and restore config from a backup if there would be still an issue? I know that's not a fix or solution...

Why i should ?

**Comment by brada4** [comment]
Run `ps ww| grep dhcp`
You get something like
`12345 root      1234 S    udhcpc -p /var/run/udhcpc-wan.pid -s /lib/netifd/dhcp.script -f -t 0 -i wan -x hostname:OpenWRT -C -R -O`
note the `-i wan` part - the interface DHCP client is querying.

Now kill the process
`kill -9 12345`

Measure "speed"
(dhcp client may come back)

`/etc/config//network` has section regarding dhcp client.
```
config interface 'wan'
        option device 'wan'
        option proto 'dhcp'
```
change to
`option proto 'none'`
to disable DHCP client. You dont need DHCP client as IP is acquired from pppd.

**Comment by Snowiiii** [comment]
> Run `ps ww| grep dhcp`
> You get something like
> `12345 root 1234 S udhcpc -p /var/run/udhcpc-wan.pid -s /lib/netifd/dhcp.script -f -t 0 -i wan -x hostname:OpenWRT -C -R -O`
> note the `-i wan` part - the interface DHCP client is querying.

i dont have an udhcpc process

![image](https://github.com/openwrt/openwrt/assets/71594357/a196aff0-1e47-424c-9dd9-bc13e1f23d67)

**Comment by brada4** [comment]
I see it in htop picture against `-I pppoe-wan`
Try to find any chance it starts by `/dhcp` in /etc/config/network.... And change to none.

**Comment by Snowiiii** [comment]
in `/etc/config/network.` my wan6 is the dhcpv6 client, I have also wan thats is pppoe
![image](https://github.com/openwrt/openwrt/assets/71594357/637b7a77-8098-48ca-825f-c4b663c83e58)

**Comment by brada4** [comment]
Weird high-cpu dhcpv4 client is in your htop
If you do not get ipv6 address you can disable (dont start at boot) the wan6.
Does not seem either is at fault in UBUS CPU usage.

Try to trace ubus for longer. Either it is infinite loop in itself or something punching same useless request at it less often than every 5s.
ubus is supposed to be low overhead ipc.

**Comment by Snowiiii** [comment]
I had tried around dhcpv4 and dhcpv6 but it had no difference. But ubus is not the Problem i looked at htop again an ubus have basicly no CPU usage

**Comment by brada4** [comment]
Strange you cannot divert route. No point in qos if bw limit is local to router.

**Comment by danpawlik** [comment]
> > Just curious: did you try to make factory reset and restore config from a backup if there would be still an issue? I know that's not a fix or solution...
> 
> Why i should ? And i just made an fresh install 3 days ago, I don't see an point for an reset

In that case there is no reason to reset to defaults.

**Comment by Snowiiii** [comment]
> Strange you cannot divert route. No point in qos if bw limit is local to router.

how what should i do now ?

**Comment by brada4** [comment]
Try to add arbitrary route via dev dsl0.7 like
`ip route add 0.1.2.3 dev dsl0.7 src 0.2.3.4`

**Comment by Snowiiii** [comment]
> Try to add arbitrary route via dev dsl0.7 like `ip route add 0.1.2.3 dev dsl0.7 src 0.2.3.4`

root@Baumhaus:~# ip route add 0.1.2.3 dev dsl0.7 src 0.2.3.4
ip: RTNETLINK answers: Invalid argument

**Comment by brada4** [comment]
See this says vlan 132....
https://forum.vodafone.de/t5/Archiv-Internet-Ger%C3%A4te/Eigene-Fritzbox-7560-nutzen-PPPoE-Pakete/m-p/1708471

**Comment by Snowiiii** [comment]
> See this says vlan 132.... https://forum.vodafone.de/t5/Archiv-Internet-Ger%C3%A4te/Eigene-Fritzbox-7560-nutzen-PPPoE-Pakete/m-p/1708471

Dosn't work, no connection at all. The Post is a bit older and back in the day this was a thing the Every Provider (In Germany) would have their own VLAN id, But today the default is ID 7, And should work for every provider at least in germany

**Comment by Snowiiii** [comment]
So slowly i more and more coming to the conclussion that its not an Config error, Its something to do with the Code, Btw PfSense has somewhat the same issue: https://www.youtube.com/watch?v=jpdQ9cstvxU, There are many people saying thats in upstream issue and the 100% CPU usage is very sus.

**Comment by Snowiiii** [comment]
btw something weird too is that when uploading the CPU seems fine but i still have an bottleneck and only getting ~1Mbit/s

**Comment by brada4** [comment]
Another (not very bright) idea:
`speedtest-netperf.sh -H netperf-eu.bufferbloat.net`

yes, low cpu seems like delay accumulating in mystery place hard to locate because it does not eat up cpu.

you may try htop show kernel threads in case you dont see any usage by user processes.

**Comment by Snowiiii** [comment]
Htop with some more details while running speed test:
![image](https://github.com/openwrt/openwrt/assets/71594357/a2dbb1ca-8338-4a3b-80a4-efa529819909)

**Comment by Snowiiii** [comment]
idk why but in the last time CPU is not maxing out anymore but is still very high, I changed nothing

**Comment by brada4** [comment]
magenta bar is softirq on 1st cpu. nstat and tc -s qdisc may shlw packet drop points. It says cpu too weak for sqm, likely even needing firewall offloads

**Comment by Snowiiii** [comment]
> idk why but since a while ago CPU is not maxing out anymore but is still very high, I changed nothing

nvm i guess haha, 
I installed `net-perf`, thats what i got:
![image](https://github.com/openwrt/openwrt/assets/71594357/080632c1-7f93-4f90-bf2b-f94f710b8af4)

**Comment by brada4** [comment]
I mean the exact script speedtest-netperf against public server.
You run client on 1 cpu server on other, they get 50MB/s around over localhost. 100x better than slow internet.

**Comment by Snowiiii** [comment]
So i currently running netperf over my router, Should i run it over my PC ?, These are the results:

```
root@Baumhaus:~# netperf -H netperf-eu.bufferbloat.net
MIGRATED TCP STREAM TEST from 0.0.0.0 (0.0) port 0 AF_INET to demo.tohojo.dk (193) port 0 AF_INET : demo
Recv   Send    Send                          
Socket Socket  Message  Elapsed              
Size   Size    Size     Time     Throughput  
bytes  bytes   bytes    secs.    10^6bits/sec  

131072  16384  16384    11.13       1.45   
```
I got 2 ssh sessions one for htop and the other for netperf

**Comment by brada4** [comment]
Full speedtest-netperf.sh script from synonymous package. I.e if it gets radically faster than what waveform sees over nat.

**Comment by brada4** [comment]
(if the speeds/latencies are same just flash back to fritzos and let your mum enjoy internets)

**Comment by brada4** [comment]
Please try speedtest-netperf script,  if it yelds no improvement over current lan speeds issue to subscribe while hetting back vendor fw is https://github.com/openwrt/openwrt/issues/10224
As you see the$e many tweaks been t$ied but non is unive$sally good and thus corrrect

**Comment by brada4** [comment]
To shake slow code paths you may create br-wan containing either dsl0 or dsl0.7 
Another is to use alternative vlan encap method akin docker-typical vxlan https://enterprise-support.nvidia.com/s/article/Configuring-VXLAN-Encap-Decap-Offload-Using-tc
The rest is pretty much same as most common gige-gige NAT setup.

**Comment by Snowiiii** [comment]
I think we need an OpenWRT coder here, Someone who can Help to Profile this stuff, Code Profiling and especially Network Profiling is a not funny Job, But im also tired of just trying random stuff, Maybe we should ping someone ?

**Comment by Snowiiii** [comment]
Okay fine, Lets investigate further when you think, so i saw in htop that 'ksoftirqd' is eating up most of the CPU usage, a quick google search showed that its an common problem
e.g. https://portal.nutanix.com/page/documents/kbs/details?targetId=kA07V000000LUR3SAO

**Comment by Snowiiii** [comment]
Btw consider that not until the Issue is fixed i can't really play Multiplayer Games anymore or Watch Youtube. So i try my best to get this fixed, Rolling back to FritzOS is not an option too me

**Comment by brada4** [comment]
Sorry, if you mix iptables-legacy with nft you get very unpredictable results.
`# conntrack -L > /dev/null`

**Comment by brada4** [comment]
iptables --version -> should say nf_tables.

**Comment by brada4** [comment]
You have to reset the configuration, set up DSL connection anew and measure speeds. Install iptables-nft if you really need iptables command line.

**Comment by Snowiiii** [comment]
So what i found https://forum.openwrt.org/t/22-03-how-to-install-iptables/147505/8 and https://forum.openwrt.org/t/warning-iptables-use-legacy/127752/4, That people just saying that `iptables-zz-legacy` like the name says it outdated and i should use `iptables-nft` instead ?

**Comment by brada4** [comment]
Unpredictable results are guaranteed if you filter using both engines at once. 
Firstly you have double hooks in unpredictable order in same places of network stack, then duplicate independent  conntrack and mark implementations, in the end you cannot tell which of two drops or rejects packet. And partial drops inside the stream guarantee abysmal speed.
I would recommend to start on clean system - connect DSL, measure speed, enable offload, measure again, enable hardware offload m.a. ... Then install iptables-nft and m.a.
....
See you in 20mins?

**Comment by Snowiiii** [comment]
okay fine, were is the reset button?...

**Comment by brada4** [comment]
Its in Luci/System/Firmware/Perform Reset

**Comment by janh** [comment]
What is the data rate of the DSL connection? It is shown on the main page of the web interface or on the command line using `ubus call dsl metrics`.

If you didn't already do it, you may need to switch to another DSL firmware than the one that is included with OpenWrt (it doesn't support vectoring, which would cause the connection to be limited to similar data rates as you are seeing). There is a [list of available firmware versions here](https://xdarklight.github.io/lantiq-xdsl-firmware-info/#VRX200). If you are unsure, pick the latest version from Fritzbox firmware, i.e. "5.9.1.4.0.7-5.9.0.D.0.2" (if the download link is broken, just use the latest Fritzbox firmware file instead).

(It is possible that you may still be limited by the CPU in the end, because OpenWrt doesn't have any hardware acceleration for routing on this platform.)

**Comment by brada4** [comment]
Software offload is 2-3x faster than simple conntrack by means of less usage by softirq hooked to NAPI interrupt. 
It skips rest of firewall processing ideally copying packet with small conversions from skb to skb.

**Comment by Snowiiii** [comment]
> What is the data rate of the DSL connection? It is shown on the main page of the web interface or on the command line using `ubus call dsl metrics`.
> 
> If you didn't already do it, you may need to switch to another DSL firmware than the one that is included with OpenWrt (it doesn't support vectoring, which would cause the connection to be limited to similar data rates as you are seeing). There is a [list of available firmware versions here](https://xdarklight.github.io/lantiq-xdsl-firmware-info/#VRX200). If you are unsure, pick the latest version from Fritzbox firmware, i.e. "5.9.1.4.0.7-5.9.0.D.0.2" (if the download link is broken, just use the latest Fritzbox firmware file instead).
> 
> However, it is possible that you will still be limited by the CPU in the end, because OpenWrt doesn't have any hardware acceleration for routing on this platform.

```
root@Baumhaus:/tmp# ubus call dsl metrics
{
	"api_version": "4.17.18.6",
	"firmware_version": "5.7.9.9.0.6",
	"chipset": "Lantiq-VRX200",
	"driver_version": "1.5.17.6",
	"state": "Showtime with TC-Layer sync",
	"state_num": 7,
	"up": true,
	"uptime": 129268,
	"atu_c": {
		"vendor_id": [
			181,
			0,
			66,
			68,
			67,
			77,
			178,
			33
		],
		"vendor": "Broadcom 178.33",
		"system_vendor_id": [
			181,
			0,
			66,
			68,
			67,
			77,
			0,
			0
		],
		"system_vendor": "Broadcom",
		"version": [
			49,
			57,
			46,
			48,
			46,
			52,
			49,
			46,
			50,
			32,
			86,
			69,
			95,
			49,
			49,
			95
		],
		"serial": [
			65,
			65,
			49,
			53,
			49,
			49,
			70,
			69,
			48,
			53,
			87,
			45,
			49,
			51,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0,
			0
		]
	},
	"power_state": "L0 - Synchronized",
	"power_state_num": 0,
	"xtse": [
		0,
		0,
		0,
		0,
		0,
		0,
		0,
		2
	],
	"annex": "B",
	"standard": "G.993.2",
	"profile": "17a",
	"mode": "G.993.2 (VDSL2, Profile 17a)",
	"upstream": {
		"vector": false,
		"trellis": true,
		"bitswap": false,
		"retx": true,
		"virtual_noise": false,
		"ra_mode": "At initialization",
		"ra_mode_num": 1,
		"interleave_delay": 0,
		"inp": 35.000000,
		"data_rate": 1577000,
		"latn": 2.600000,
		"satn": 2.600000,
		"snr": 8.800000,
		"actatp": 4.300000,
		"attndr": 1577000,
		"mineftr": 1569000
	},
	"downstream": {
		"vector": false,
		"trellis": true,
		"bitswap": true,
		"retx": true,
		"virtual_noise": false,
		"ra_mode": "At initialization",
		"ra_mode_num": 1,
		"interleave_delay": 730,
		"inp": 67.000000,
		"data_rate": 19145000,
		"latn": 6.500000,
		"satn": 6.500000,
		"snr": 5.900000,
		"actatp": 13.600000,
		"attndr": 19117788,
		"mineftr": 18948000
	},
	"olr": {
		"downstream": {
			"bitswap": {
				"requested": 62,
				"executed": 31,
				"rejected": 0,
				"timeout": 0
			},
			"sra": {
				"requested": 0,
				"executed": 0,
				"rejected": 0,
				"timeout": 0
			},
			"sos": {
				"requested": 0,
				"executed": 0,
				"rejected": 0,
				"timeout": 0
			}
		},
		"upstream": {
			"bitswap": {
				"requested": 0,
				"executed": 0,
				"rejected": 0,
				"timeout": 0
			},
			"sra": {
				"requested": 0,
				"executed": 0,
				"rejected": 0,
				"timeout": 0
			},
			"sos": {
				"requested": 0,
				"executed": 0,
				"rejected": 0,
				"timeout": 0
			}
		}
	},
	"errors": {
		"near": {
			"es": 0,
			"ses": 0,
			"loss": 0,
			"uas": 90,
			"lofs": 0,
			"fecs": 0,
			"leftrs": 0,
			"cv_c": 0,
			"fec_c": 0,
			"hec": 0,
			"ibe": 0,
			"crc_p": 0,
			"crcp_p": 0,
			"cv_p": 0,
			"cvp_p": 0,
			"rx_corrupted": 1009,
			"rx_uncorrected_protected": 0,
			"rx_retransmitted": 0,
			"rx_corrected": 1009,
			"tx_retransmitted": 19
		},
		"far": {
			"es": 357,
			"ses": 0,
			"loss": 0,
			"uas": 90,
			"lofs": 0,
			"fecs": 2934,
			"leftrs": 65,
			"cv_c": 371,
			"fec_c": 6632,
			"hec": 0,
			"ibe": 0,
			"crc_p": 0,
			"crcp_p": 0,
			"cv_p": 0,
			"cvp_p": 0,
			"rx_corrupted": 121996,
			"rx_uncorrected_protected": 104750,
			"rx_retransmitted": 0,
			"rx_corrected": 17246,
			"tx_retransmitted": 21747
		}
	}
}
```

**Comment by brada4** [comment]
See data rate 20/1.5mbps, yes need new firmware to support physical medium.

**Comment by Snowiiii** [comment]
> See data rate 20/1.5mbps, yes need new firmware to support physical medium.

Okay i just perform a reset to be safe, Now im ready to to upgrade firmware, The download link is broken but i found one on the avm site: https://download.avm.de/fritzbox/fritzbox-7430/deutschland/fritz.os/FRITZ.Box_7430-07.31.image

**Comment by Snowiiii** [comment]
what commands i should use for some reason cloning https://github.com/Freetz-NG/freetz-ng using git or git-http resulting in an error

```
root@Baumhaus:/tmp#  git clone https://github.com/Freetz-NG/freetz-ng
Cloning into 'freetz-ng'...
remote: Enumerating objects: 224085, done.
remote: Counting objects: 100% (1433/1433), done.
remote: Compressing objects: 100% (603/603), done.
error: 2135 bytes of body are still expected0 MiB | 34.00 KiB/s 
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
```

**Comment by brada4** [comment]
It is 200MB repo, your router does not have space, clne or download raw firmware bin to your desktop.

**Comment by Snowiiii** [comment]
Is there are way to compile this on arch, It says:

```
Unsupported, file for your Linux version not found: Manjaro

	Available names for 'linux+version' parameter:
	  Debian
	  Debian11
	  Debian12
	  Devuan
	  Devuan4
	  Devuan5
	  Fedora
	  Fedora20
	  Fedora33
	  Fedora34
	  Fedora35
	  Fedora36
	  Fedora37
	  Fedora38
	  Fedora39
	  Mint
	  Mint17
	  Mint18
	  Mint19
	  Mint20
	  Mint21
	  Ubuntu
	  Ubuntu09
	  Ubuntu10
	  Ubuntu11
	  Ubuntu12
	  Ubuntu13
	  Ubuntu14
	  Ubuntu15
	  Ubuntu16
	  Ubuntu17
	  Ubuntu18
	  Ubuntu19
	  Ubuntu20
	  Ubuntu21
	  Ubuntu22
	  Ubuntu23
```

**Comment by brada4** [comment]
They provide documentation, binaries and containers.

**Comment by Snowiiii** [comment]
> They provide documentation, binaries and containers.

Okay i installed Ubuntu VM, And executeded all the commands, What file i need now ?
![image](https://github.com/openwrt/openwrt/assets/71594357/ebba87c1-1404-49f2-8482-c0df3af20ca8)

**Comment by Snowiiii** [comment]
> @Snowiiii maybe that will help you: https://github.com/danpawlik/fritzbox-custom-configs . Long time not used. Might be outdated.

nah its not what i need but thanks

**Comment by brada4** [comment]
Check dmesg, it loads some file in /lib/firmware , not finding that falls back to file named vdsl.bin
Just check "data_rate" after you bring up dsl connection.

**Comment by Snowiiii** [comment]
``` 
root@Baumhaus:/tmp# dmesg
[    0.000000] Linux version 5.15.137 (builder@buildhost) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r23630-842932a63d) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP Tue Nov 14 13:38:11 2023
[    0.000000] SoC: xRX200 rev 1.2
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019556 (MIPS 34Kc)
[    0.000000] MIPS: machine is AVM FRITZ!Box 7430
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 10 pages/cpu s12240 r8192 d20528 u40960
[    0.000000] pcpu-alloc: s12240 r8192 d20528 u40960 alloc=10*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: console=ttyLTQ0,115200
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=00069068
[    0.000000] Readback ErrCtl register=00069068
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 119204K/131072K available (7224K kernel code, 629K rwdata, 1156K rodata, 1212K init, 220K bss, 11868K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] CPU Clock: 500MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041786 ns
[    0.000002] sched_clock: 32 bits at 250MHz, resolution 4ns, wraps every 8589934590ns
[    0.008001] Calibrating delay loop... 331.77 BogoMIPS (lpj=1658880)
[    0.074089] pid_max: default: 32768 minimum: 301
[    0.080206] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.087435] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.101645] rcu: Hierarchical SRCU implementation.
[    0.107476] smp: Bringing up secondary CPUs ...
[    0.113487] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.113529] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.113614] CPU1 revision is: 00019556 (MIPS 34Kc)
[    0.167809] Synchronize counters for CPU 1: done.
[    0.192106] smp: Brought up 1 node, 2 CPUs
[    0.205743] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.215536] futex hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.222595] pinctrl core: initialized pinctrl subsystem
[    0.230587] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.237458] thermal_sys: Registered thermal governor 'step_wise'
[    0.242037] dcdc-xrx200 1f106a00.dcdc: Core Voltage : 1016 mV
[    0.261796] pinctrl-xway 1e100b10.pinmux: Init done
[    0.268127] dma-xway 1e104100.dma: Init done - hw rev: 7, ports: 7, channels: 28
[    0.279479] pcie-xrx200 1d900000.pcie: failed to get the PCIe PHY
[    0.309439] usbcore: registered new interface driver usbfs
[    0.314876] usbcore: registered new interface driver hub
[    0.320442] usbcore: registered new device driver usb
[    0.328629] clocksource: Switched to clocksource MIPS
[    0.336198] NET: Registered PF_INET protocol family
[    0.341449] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.350141] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.358393] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.366297] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.374012] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.381117] TCP: Hash tables configured (established 1024 bind 1024)
[    0.387676] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.394184] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.401714] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.407302] PCI: CLS 0 bytes, default 32
[    0.411382] gptu: totally 6 16-bit timers/counters
[    0.416279] gptu: misc_register on minor 127
[    0.420731] gptu: succeeded to request irq 126
[    0.425076] gptu: succeeded to request irq 127
[    0.429762] gptu: succeeded to request irq 128
[    0.434294] gptu: succeeded to request irq 129
[    0.438732] gptu: succeeded to request irq 130
[    0.443153] gptu: succeeded to request irq 131
[    0.447867] No VPEs reserved for AP/SP, not initialize VPE loader
[    0.447867] Pass maxvpes=<n> argument as kernel argument
[    0.459302] No TCs reserved for AP/SP, not initializing RTLX.
[    0.459302] Pass maxtcs=<n> argument as kernel argument
[    0.473860] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.491695] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.497389] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.509970] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.524777] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    0.533809] printk: console [ttyLTQ0] enabled
[    0.542522] printk: bootconsole [early0] disabled
[    0.557437] nand: device found, Manufacturer ID: 0x98, Chip ID: 0xf1
[    0.562387] nand: Toshiba NAND 128MiB 3,3V 8-bit
[    0.566991] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.574902] nand: WARNING: 10000000.flash: the ECC used on your system (1b/256B) is too weak compared to the one required by the NAND chip (8b/512B)
[    0.588001] Scanning device for bad blocks
[    0.675952] Bad eraseblock 768 at 0x000006000000
[    0.706628] 8 fixed-partitions partitions found on MTD device 10000000.flash
[    0.712310] Creating 8 MTD partitions on "10000000.flash":
[    0.717735] 0x000000000000-0x000000040000 : "urlader"
[    0.724446] 0x000000040000-0x000000440000 : "nand-tffs"
[    0.730269] 0x000000440000-0x000000840000 : "kernel"
[    0.735573] 0x000000840000-0x000003840000 : "ubi"
[    0.743149] 0x000003840000-0x000003c40000 : "reserved-kernel"
[    0.749387] 0x000003c40000-0x000006c40000 : "reserved-filesystem"
[    0.756586] 0x000006c40000-0x000006e40000 : "config"
[    0.763305] 0x000006e40000-0x000008000000 : "nand-filesystem"
[    1.550474] gswip 1e108000.switch: dsa switch register failed: -517
[    1.563103] hwmon hwmon0: temp1_input not attached to any thermal zone
[    1.576409] NET: Registered PF_INET6 protocol family
[    1.593370] Segment Routing with IPv6
[    1.595761] In-situ OAM (IOAM) with IPv6
[    1.599808] NET: Registered PF_PACKET protocol family
[    1.604733] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    1.618729] 8021q: 802.1Q VLAN Support v1.8
[    1.627789] pcie-xrx200 1d900000.pcie: failed to get the PCIe PHY
[    2.409082] gswip 1e108000.switch: Unsupported port: 6
[    2.412927] gswip 1e108000.switch: Unsupported port: 6
[    2.417954] gswip 1e108000.switch: fixed link full duplex 1000Mbps not recognised
[    2.425556] gswip 1e108000.switch: configuring for fixed/internal link mode
[    2.432844] gswip 1e108000.switch: Link is Up - 1Gbps/Full - flow control off
[    2.441953] gswip 1e108000.switch lan3 (uninitialized): PHY [1e108000.switch-mii:11] driver [Intel XWAY PHY22F (xRX v1.2 integrated)] (irq=POLL)
[    2.460609] gswip 1e108000.switch lan4 (uninitialized): PHY [1e108000.switch-mii:12] driver [Intel XWAY PHY22F (xRX v1.2 integrated)] (irq=POLL)
[    2.478046] gswip 1e108000.switch lan1 (uninitialized): PHY [1e108000.switch-mii:13] driver [Intel XWAY PHY22F (xRX v1.2 integrated)] (irq=POLL)
[    2.495664] gswip 1e108000.switch lan2 (uninitialized): PHY [1e108000.switch-mii:14] driver [Intel XWAY PHY22F (xRX v1.2 integrated)] (irq=POLL)
[    2.509380] DSA: tree 0 setup
[    2.510966] gswip 1e108000.switch: probed GSWIP version 21 mod 0
[    2.642348] PCI host bridge to bus 0000:00
[    2.645061] pci_bus 0000:00: root bus resource [mem 0x1c000000-0x1cffffff]
[    2.651946] pci_bus 0000:00: root bus resource [io  0x1d800000-0x1d8fffff]
[    2.658834] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    2.666755] ifx_pcie_rc_class_early_fixup: fixed pcie host bridge to pci-pci bridge
[    2.684446] pci 0000:00:00.0: 0x80707828 took 17267 usecs
[    2.688422] pci 0000:00:00.0: [1bef:0011] type 01 class 0x060400
[    2.694603] pci 0000:00:00.0: PME# supported from D0 D3hot
[    2.703849] pci 0000:01:00.0: [168c:abcd] type 00 class 0x020000
[    2.708603] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x0001ffff 64bit]
[    2.715309] pci 0000:01:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    2.722153] pci 0000:01:00.0: supports D1
[    2.725943] pci 0000:01:00.0: PME# supported from D0 D1 D3hot
[    2.735060] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    2.740324] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 01
[    2.746908] pci 0000:00:00.0: BAR 8: assigned [mem 0x1c000000-0x1c0fffff]
[    2.753731] pci 0000:00:00.0: BAR 9: assigned [mem 0x1c100000-0x1c1fffff pref]
[    2.760949] pci 0000:01:00.0: BAR 0: assigned [mem 0x1c000000-0x1c01ffff 64bit]
[    2.768233] pci 0000:01:00.0: BAR 6: assigned [mem 0x1c100000-0x1c10ffff pref]
[    2.775482] pci 0000:00:00.0: PCI bridge to [bus 01]
[    2.780451] pci 0000:00:00.0:   bridge window [mem 0x1c000000-0x1c0fffff]
[    2.787178] pci 0000:00:00.0:   bridge window [mem 0x1c100000-0x1c1fffff pref]
[    2.794586] ifx_pcie_bios_map_irq port 0 dev 0000:00:00.0 slot 0 pin 1 
[    2.801051] ifx_pcie_bios_map_irq dev 0000:00:00.0 irq 144 assigned
[    2.807304] pcieport 0000:00:00.0: enabling device (0000 -> 0002)
[    2.817251] UBI: auto-attach mtd3
[    2.819267] ubi0: attaching mtd3
[    2.897143] ubi0: scanning is finished
[    2.913213] ubi0: attached mtd3 (name "ubi", size 48 MiB)
[    2.917242] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 129024 bytes
[    2.924179] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 512
[    2.930902] ubi0: VID header offset: 512 (aligned 512), data offset: 2048
[    2.937580] ubi0: good PEBs: 384, bad PEBs: 0, corrupted PEBs: 0
[    2.943650] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
[    2.950868] ubi0: max/mean erase counter: 2/0, WL threshold: 4096, image sequence number: 1109729045
[    2.959996] ubi0: available PEBs: 0, total reserved PEBs: 384, PEBs reserved for bad PEB handling: 20
[    2.969664] ubi0: background thread "ubi_bgt0d" started, PID 261
[    2.976390] block ubiblock0_0: created from ubi0:0(rootfs)
[    2.980802] ubiblock: device ubiblock0_0 (rootfs) set to be root filesystem
[    2.999419] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    3.006880] Freeing unused kernel image (initmem) memory: 1212K
[    3.011441] This architecture does not have kernel memory protection.
[    3.017807] Run /sbin/init as init process
[    3.021949]   with arguments:
[    3.021971]     /sbin/init
[    3.021981]   with environment:
[    3.021989]     HOME=/
[    3.021997]     TERM=linux
[    3.636899] init: Console is alive
[    3.639615] init: - watchdog -
[    3.642323] init: Watchdog has previously reset the system
[    5.052644] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    5.131496] dwc2 1e101000.usb: supply vusb_d not found, using dummy regulator
[    5.137995] dwc2 1e101000.usb: supply vusb_a not found, using dummy regulator
[    5.247154] dwc2 1e101000.usb: DWC OTG Controller
[    5.250598] dwc2 1e101000.usb: new USB bus registered, assigned bus number 1
[    5.257577] dwc2 1e101000.usb: irq 62, io mem 0x1e101000
[    5.264779] hub 1-0:1.0: USB hub found
[    5.267373] hub 1-0:1.0: 1 port detected
[    5.275519] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    5.292226] init: - preinit -
[    6.690651] random: jshn: uninitialized urandom read (4 bytes read)
[    6.951279] random: jshn: uninitialized urandom read (4 bytes read)
[    8.998236] random: jshn: uninitialized urandom read (4 bytes read)
[    9.110608] random: jshn: uninitialized urandom read (4 bytes read)
[    9.213165] random: jshn: uninitialized urandom read (4 bytes read)
[    9.554270] gswip 1e108000.switch lan1: configuring for phy/internal link mode
[   11.854746] UBIFS (ubi0:1): Mounting in unauthenticated mode
[   11.859530] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 402
[   11.917319] UBIFS (ubi0:1): recovery needed
[   12.075704] UBIFS (ubi0:1): recovery completed
[   12.079392] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[   12.086572] UBIFS (ubi0:1): LEB size: 129024 bytes (126 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   12.096547] UBIFS (ubi0:1): FS size: 40384512 bytes (38 MiB, 313 LEBs), max 323 LEBs, journal size 2064384 bytes (1 MiB, 16 LEBs)
[   12.108190] UBIFS (ubi0:1): reserved for root: 1907459 bytes (1862 KiB)
[   12.114844] UBIFS (ubi0:1): media format: w5/r0 (latest is w5/r0), UUID 8223C3FD-1FCD-4531-B450-6B9B43C69C03, small LPT model
[   12.130694] mount_root: switching to ubifs overlay
[   12.149612] urandom-seed: Seeding with /etc/urandom.seed
[   12.313660] procd: - early -
[   12.315514] procd: - watchdog -
[   12.318854] procd: Watchdog has previously reset the system
[   13.011741] procd: - watchdog -
[   13.013655] procd: Watchdog has previously reset the system
[   13.020727] procd: - ubus -
[   13.115880] random: ubusd: uninitialized urandom read (4 bytes read)
[   13.125239] random: ubusd: uninitialized urandom read (4 bytes read)
[   13.136695] random: ubusd: uninitialized urandom read (4 bytes read)
[   13.155877] procd: - init -
[   14.203543] random: jshn: uninitialized urandom read (4 bytes read)
[   14.229251] random: ubusd: uninitialized urandom read (4 bytes read)
[   14.489411] kmodloader: loading kernel modules from /etc/modules.d/*
[   15.118958] IFXOS, Version 1.7.1 (c) Copyright 2009, Lantiq Deutschland GmbH
[   15.131594] NET: Registered PF_ATMPVC protocol family
[   15.135239] NET: Registered PF_ATMSVC protocol family
[   15.152916] PPP generic driver version 2.4.2
[   15.181254] Lantiq (VRX) DSL CPE MEI driver, version 1.5.17.6, (c) 2007-2015 Lantiq Beteiligungs-GmbH & Co. KG
[   15.197957] 
[   15.197957] 
[   15.197957] Lantiq CPE API Driver version: DSL CPE API V4.17.18.6
[   15.215721] 
[   15.215721] Predefined debug level: 3
[   15.241206] Loading modules backported from Linux version v6.1.24-0-g0102425ac76b
[   15.247285] Backport generated by backports.git v5.15.92-1-44-gd6ea70fafd36
[   15.264069] Infineon Technologies DEU driver version 2.0.0 
[   15.272958] IFX DEU DES initialized (multiblock).
[   15.283225] IFX DEU AES initialized (multiblock).
[   15.287197] IFX DEU SHA1 initialized.
[   15.291011] IFX DEU MD5 initialized.
[   15.294418] IFX DEU SHA1_HMAC initialized.
[   15.298594] IFX DEU MD5_HMAC initialized.
[   15.329025] NET: Registered PF_PPPOX protocol family
[   15.929211] ifx_pcie_bios_map_irq port 0 dev 0000:01:00.0 slot 0 pin 1 
[   15.934459] ifx_pcie_bios_map_irq dev 0000:01:00.0 irq 144 assigned
[   15.940909] ath9k 0000:01:00.0: enabling device (0000 -> 0002)
[   15.955597] ath: EEPROM regdomain: 0x0
[   15.955650] ath: EEPROM indicates default country code should be used
[   15.955667] ath: doing EEPROM country->regdmn map search
[   15.955690] ath: country maps to regdmn code: 0x3a
[   15.955709] ath: Country alpha2 being used: US
[   15.955727] ath: Regpair used: 0x3a
[   15.973305] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   15.979922] ieee80211 phy0: Atheros AR9300 Rev:3 mem=0xbc000000, irq=144
[   15.998233] kmodloader: done loading kernel modules from /etc/modules.d/*
[   19.296269] urngd: v1.0.2 started.
[   22.980682] random: crng init done
[   22.982686] random: 69 urandom warning(s) missed due to ratelimiting
[   39.570067] device eth0 entered promiscuous mode
[   39.573438] gswip 1e108000.switch lan1: configuring for phy/internal link mode
[   39.591103] br-lan: port 1(lan1) entered blocking state
[   39.594936] br-lan: port 1(lan1) entered disabled state
[   39.615977] device lan1 entered promiscuous mode
[   39.625908] gswip 1e108000.switch: port 4 failed to add 1e:4c:aa:6f:94:08 vid 1 to fdb: -22
[   39.638731] gswip 1e108000.switch: port 4 failed to add 7c:ff:4d:49:c2:c0 vid 0 to fdb: -22
[   39.645804] gswip 1e108000.switch: port 4 failed to add 7c:ff:4d:49:c2:c0 vid 1 to fdb: -22
[   39.677766] gswip 1e108000.switch: port 4 failed to delete 1e:4c:aa:6f:94:08 vid 1 from fdb: -2
[   39.710453] gswip 1e108000.switch lan2: configuring for phy/internal link mode
[   39.729241] br-lan: port 2(lan2) entered blocking state
[   39.733077] br-lan: port 2(lan2) entered disabled state
[   39.790976] device lan2 entered promiscuous mode
[   39.796915] gswip 1e108000.switch: port 5 failed to add 7c:ff:4d:49:c2:c0 vid 1 to fdb: -22
[   39.809557] gswip 1e108000.switch: port 5 failed to add 7c:ff:4d:49:c2:c0 vid 0 to fdb: -22
[   39.838528] gswip 1e108000.switch lan3: configuring for phy/internal link mode
[   39.847602] br-lan: port 3(lan3) entered blocking state
[   39.851629] br-lan: port 3(lan3) entered disabled state
[   39.872327] device lan3 entered promiscuous mode
[   39.887845] gswip 1e108000.switch: port 2 failed to add 7c:ff:4d:49:c2:c0 vid 1 to fdb: -22
[   39.895016] gswip 1e108000.switch: port 2 failed to add 7c:ff:4d:49:c2:c0 vid 0 to fdb: -22
[   40.008364] gswip 1e108000.switch lan4: configuring for phy/internal link mode
[   40.019415] br-lan: port 4(lan4) entered blocking state
[   40.023248] br-lan: port 4(lan4) entered disabled state
[   40.057000] device lan4 entered promiscuous mode
[   40.065880] gswip 1e108000.switch: port 3 failed to add 7c:ff:4d:49:c2:c0 vid 1 to fdb: -22
[   40.074067] gswip 1e108000.switch: port 3 failed to add 7c:ff:4d:49:c2:c0 vid 0 to fdb: -22
[   41.901693] ath: EEPROM regdomain: 0x8114
[   41.904326] ath: EEPROM indicates we should expect a country code
[   41.910609] ath: doing EEPROM country->regdmn map search
[   41.915694] ath: country maps to regdmn code: 0x37
[   41.920603] ath: Country alpha2 being used: DE
[   41.924920] ath: Regpair used: 0x37
[   41.928401] ath: regdomain 0x8114 dynamically updated by user
[   42.749193] gswip 1e108000.switch lan1: Link is Up - 100Mbps/Full - flow control rx/tx
[   42.755775] br-lan: port 1(lan1) entered blocking state
[   42.761068] br-lan: port 1(lan1) entered forwarding state
[   42.788720] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   47.726315] br-lan: port 5(phy0-ap0) entered blocking state
[   47.730623] br-lan: port 5(phy0-ap0) entered disabled state
[   47.736763] device phy0-ap0 entered promiscuous mode
[   47.741946] gswip 1e108000.switch: port 4 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   47.758715] gswip 1e108000.switch: port 5 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   47.765742] gswip 1e108000.switch: port 2 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   47.808662] gswip 1e108000.switch: port 3 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   47.815682] gswip 1e108000.switch: port 4 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   47.867574] gswip 1e108000.switch: port 5 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   47.881504] device phy0-ap0 left promiscuous mode
[   47.885065] br-lan: port 5(phy0-ap0) entered disabled state
[   47.926733] br-lan: port 5(phy0-ap0) entered blocking state
[   47.931090] br-lan: port 5(phy0-ap0) entered disabled state
[   47.937290] device phy0-ap0 entered promiscuous mode
[   47.948721] gswip 1e108000.switch: port 2 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   47.955755] gswip 1e108000.switch: port 3 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   47.996033] gswip 1e108000.switch: port 4 failed to delete 00:02:03:04:05:06 vid 1 from fdb: -2
[   48.027105] gswip 1e108000.switch: port 5 failed to delete 00:02:03:04:05:06 vid 1 from fdb: -2
[   48.058838] gswip 1e108000.switch: port 2 failed to delete 00:02:03:04:05:06 vid 1 from fdb: -2
[   48.066206] gswip 1e108000.switch: port 3 failed to delete 00:02:03:04:05:06 vid 1 from fdb: -2
[   48.103626] gswip 1e108000.switch: port 4 failed to delete 00:02:03:04:05:06 vid 0 from fdb: -2
[   48.128760] gswip 1e108000.switch: port 5 failed to delete 00:02:03:04:05:06 vid 0 from fdb: -2
[   48.136102] gswip 1e108000.switch: port 2 failed to delete 00:02:03:04:05:06 vid 0 from fdb: -2
[   48.208400] gswip 1e108000.switch: port 3 failed to delete 00:02:03:04:05:06 vid 0 from fdb: -2
[   48.215917] gswip 1e108000.switch: port 4 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   48.348697] gswip 1e108000.switch: port 5 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   48.355775] gswip 1e108000.switch: port 2 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   48.418779] gswip 1e108000.switch: port 3 failed to add 00:02:03:04:05:06 vid 0 to fdb: -22
[   48.425854] gswip 1e108000.switch: port 4 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   48.488638] gswip 1e108000.switch: port 5 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   48.518751] gswip 1e108000.switch: port 2 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   48.525847] gswip 1e108000.switch: port 3 failed to add 00:02:03:04:05:06 vid 1 to fdb: -22
[   56.194383] IPv6: ADDRCONF(NETDEV_CHANGE): phy0-ap0: link becomes ready
[   56.200055] br-lan: port 5(phy0-ap0) entered blocking state
[   56.205202] br-lan: port 5(phy0-ap0) entered forwarding state
[   80.555145] Lantiq PTM driver, version 1.0.27, (c) 2001-2013 Lantiq Deutschland GmbH
[   80.555145]     PTM (E1) firmware version 2.16.0
[   80.566179] ifxmips_ptm: PTM init succeed
[  125.232838] enter showtime
[  125.234483] IPv6: ADDRCONF(NETDEV_CHANGE): dsl0: link becomes ready
[  125.244638] IPv6: ADDRCONF(NETDEV_CHANGE): dsl0.7: link becomes ready
[  125.258296] enter showtime
[  126.228658] pppoe-wan: renamed from ppp0
```

**Comment by Snowiiii** [comment]
i just see 
```
Lantiq PTM driver, version 1.0.27, (c) 2001-2013 Lantiq Deutschland GmbH
[   80.555145]     PTM (E1) firmware version 2.16.0
```

**Comment by brada4** [comment]
Upload bin file to /lib/firmware under whatever name and point luci network interface to "custom firmware file"

**Comment by Snowiiii** [comment]
I have, Thats all the files https://www.mediafire.com/folder/wtldea4jsn0s4k3,hpbuv8mk4hprtno,y3yybfnhdg79b8s/shared

vr9-A-dsl did not work, i put in `lib/firmware/vr9-A-dsl.bin` into luci

**Comment by Snowiiii** [comment]
I think i did something wrong, Internet speed is not changing and i dont see any firmware loaded log
![image](https://github.com/openwrt/openwrt/assets/71594357/7e7854fa-8f59-485a-9fcf-7af767b067fc)

**Comment by KanjiMonster** [comment]
You have selected Annex B, so you probably want the `vr9-B-dsl.bin`, not `vr9-A-dsl.bin`. guessing from the filenames.

**Comment by Snowiiii** [comment]
> You have selected Annex B, so you probably want the `vr9-B-dsl.bin`, not `vr9-A-dsl.bin`. guessing from the filenames.

idk really the diffrence and idk what i need or want

**Comment by Snowiiii** [comment]
Okay i switched to the other file i guess:
![image](https://github.com/openwrt/openwrt/assets/71594357/bd4bcfec-747f-4baa-9efd-c34c41f93931)
still bad internet

**Comment by brada4** [comment]
Firmware version is upgraded, the VM succeeded at that. Try different modes in top box, it says "vector:false" , you need them to exceed standard rates of Annex B

**Comment by janh** [comment]
Maybe your line has gotten stuck in the non-vectoring fallback profile. If that is the case, you'll need to contact your ISP to reset it.

Trying different modes is very unlikely to help, but if you try it anyway, only the two Annex B + VDSL2 modes make sense at all.

**Comment by Snowiiii** [comment]
Switched from Annex B -> Annex B/J
No difference 
![image](https://github.com/openwrt/openwrt/assets/71594357/2e8e70fe-a058-487c-855d-23d160c113a7)

**Comment by Snowiiii** [comment]
> Disconnect wall socket, wait for 3am to restart pbx config etc...

-_-

**Comment by brada4** [comment]
No tricks, disconnect phone socket for few minutes now and try again.
Then in the morning try another time.

**Comment by Snowiiii** [comment]
What do you even mean by wall socket or phone socket

**Comment by brada4** [comment]
The line leading to your provider....

**Comment by Snowiiii** [comment]
so you say i should physically remove the DSL cable ?
> Disconnect wall socket, wait for 3am to restart pbx config etc...
i tought you mean 3am night

**Comment by brada4** [comment]
Unplugging cable may cause line to reset.
If not it may kick in when provider re-programs exchange you are connected to (or it may not and you call them tomorrow)

**Comment by janh** [comment]
If vectoring is used, then all lines in the same cable bundle have to use it, no matter the data rate. Otherwise it would negatively affect the lines using vectoring (as they would not be able to compensate for crosstalk originating from your line).

The only exception here in Germany is that the frequency range up to 2.2 MHz (the first 512 carriers) is excluded, as it is also used for ADSL which doesn't support vectoring anyway. That is also how the non-vectoring fallback profile works, by the way. It still uses VDSL2, but limited to only that frequency range.

If you install the package `luci-mod-dsl` (which adds a new DSL entry to the status menu of the web interface) and then look at the spectrum graphs, you should also be able to see this frequency limitation.

**Comment by Snowiiii** [comment]
![image](https://github.com/openwrt/openwrt/assets/71594357/8df56949-6c2a-415d-bf87-d1d5ea32acf0)
![image](https://github.com/openwrt/openwrt/assets/71594357/e721a20e-08e4-4e76-9f8b-04d1eab62df7)
![image](https://github.com/openwrt/openwrt/assets/71594357/3a631f72-7e5d-4fed-9507-8e8e499ff9eb)

**Comment by janh** [comment]
That looks exactly as expected for the non-vectoring fallback profile.

Personally, I don't think you'll be able to avoid calling your ISP in this case. The only other thing still worth trying might be to revert to vendor firmware for a test.

**Comment by Snowiiii** [comment]
> That looks exactly as expected for the non-vectoring fallback profile.
> 
> Personally, I don't think you'll be able to avoid calling your ISP in this case. The only other thing still worth trying might be to revert to vendor firmware for a test.

nah i dont think that meine nerven dafür reichen, I already spend to much time installing openwrt configuring it reseting it and now going back would be an not funny, But i question me how can i be that as a had FritzOS everything was fine and then going to OpenWRT instantly an internet drop i dont think that this has to do with my ISP

**Comment by brada4** [comment]
If fritzos works the firmware did not work out. Thats best you can test before calling.

**Comment by janh** [comment]
> But i question me how can i be that as a had FritzOS everything was fine and then going to OpenWRT instantly an internet drop

The DSL firmware included with OpenWrt doesn't support vectoring, so it is normal that the line runs in the non-vectoring fallback profile at first. However, usually it restores itself automatically after switching to the proper firmware. For some reason, that doesn't happen in your case, and the line is still stuck in the fallback, so a manual reset is required.

**Comment by Snowiiii** [comment]
> > But i question me how can i be that as a had FritzOS everything was fine and then going to OpenWRT instantly an internet drop
> 
> The DSL firmware included with OpenWrt doesn't support vectoring, so it is normal that the line runs in the non-vectoring fallback profile (with lower data rate) at first. However, usually it restores itself automatically after switching to the proper firmware. For some reason, that doesn't happen in your case, and the line is still stuck in the fallback, so a manual reset is required.

what are the options for an "manuel reset" ?, besides calling my ISP

**Comment by brada4** [comment]
1/ wait for next morning, if you have worst line in village exchange they may reset it automatically.
2/ revert to fritzos, it  may include some magic that shakes phone line better.
Neither is guaranteed.

**Comment by Snowiiii** [comment]
wow these options..., I dont think that something will change when i wake up next morning, I mean i have the same speed results for 4 days now the firmware change has changed also nothing

**Comment by brada4** [comment]
They keep line training data in other end so that reconnection does not yield hour of error-triggered re-trains.

**Comment by Snowiiii** [comment]
Okay we have the next day nothing changed so far speed is still bad, But i actually called my ISP today and upgraded from 50->100Mbit/s download and from 10->40Mbit/s upload, They will send me an new router to the Fritz!Box 7530, It is ARM based has an entire different CPU and +2 more CPU cores.

**Comment by brada4** [comment]
prepare dsl firmware if you intend to flash openwrt.

**Comment by Snowiiii** [comment]
> prepare dsl firmware if you intend to flash openwrt.

bro i flashed it here it did nothing

**Comment by brada4** [comment]
xref https://forum.openwrt.org/t/no-more-vdsl-vectoring-after-upgrading-from-22-03-to-23-05/197071

**Comment by Snowiiii** [comment]
> xref https://forum.openwrt.org/t/no-more-vdsl-vectoring-after-upgrading-from-22-03-to-23-05/197071
> 
> EDIT: in short vectoring FW used to work in v22, but vectoring disappeared in V23.

Regressions are so great,

![image](https://github.com/openwrt/openwrt/assets/71594357/96233522-95d9-4ab9-b782-aba4d9ba2ab2)

**Comment by tome9111991** [comment]
hi, i have a 7520 and also bad performance.
i tried latest snapshoot and 23.05.4

i'm connected via wifi 5ghz
![Screenshot 2024-08-15 110258](https://github.com/user-attachments/assets/e0c3c044-cef6-4181-ab8c-cb2f6503cff8)

But there are also days when it works well.
sometimes DSL is slow or I often have these ping spikes

**Comment by Djfe** [comment]
the FB7430 doesn't have 5GHz

**Comment by brada4** [comment]
@tome9111991 what are you trying to emphasize?

**Comment by Neustradamus** [comment]
Note that PPP 2.5.1 has been released and has been added in OpenWrt.

Snowiiii: Maybe good to update your old and unsecure PPP 2.4.2 that I see in log file.

**Comment by stephanfriedrich** [comment]
Hi at All 🌈 and 🌟 @Snowiiii,
i have had the same problem of poor latency for some time now. i thought it was due to my vdsl setup, but that is not the case.

-----

i was able to fix this with the **following ✨ sqm setup ✨**  ❤️

in your case, the sqm setup may not have been correct.
at first it did not work for me on the wrong interface either. in your case `dsl0` is certainly important and correct

**maybe this will help you too.**

```MD
1. `Network => SQM QoS => Basic Setting`
    - enable SQM instance
    - set `interface name` to your WAN-internet link _(`dsl0`, `Network → Interfaces`)_
    - set `Download speed (ingress)` to `90%` of the results you tested in Preparation
    - set `Upload speed (egress)` also to `90%` of your dsl-speed
2. `Network => SQM QoS => Queue Discipline`
    - `Queueing discipline` to `cake`
    - `Queue setup script` to `piece_of_cake.qos`
3. `Network => SQM QoS => Link Layer Adaptation`
    - select your link and overhead (setting mpu is optional see note 2):
        - `VDSL` - choose Ethernet, and set overhead 34 (or 26 if you're not using PPPoE) (mpu 68)
           If the link is 100 Mbps Ethernet set overhead 42 (mpu 84)
        - If unsure - it's better to overestimate, choose Ethernet, and set overhead 44 (mpu 96)
4. Click Save & Apply
5. Done!
    - You can confirm results by re-running the speedtest
    - Any increased ping during download/uploads will now be minimal
```

My sources for this:

- [openwrt_sqm](https://openwrt.org/docs/guide-user/network/traffic-shaping/sqm)
- [speedtest-bufferbloat](https://www.waveform.com/tools/bufferbloat)
 
-----

addendum:
i have tested the latency here on a fresh openwrt with and sqm setup (as of 06-2025)
  - **openwrt v22** (vdsl modules like “kmod-ltq-vdsl-vr9”)
     - without sqm, there are also latency problems here
     - correct sqm-setup fixes latency problems
  - **openwrt v24**
     - without sqm, latency problems
     ![Image](https://github.com/user-attachments/assets/fee48d95-7752-4801-83c2-daa5325f114e)
     - correct sqm-setup fixes latency problems
       ![Image](https://github.com/user-attachments/assets/f180bc3a-f799-4cec-8b0f-2305cf130058)


-----

have a nice lovely 🌞 day

**Comment by stephanfriedrich** [comment]
hello there,

i just have another problem _(for openwrt v24)_ that the sqm-service does not start correctly after the reboot.
(@Snowiiii, maybe that's the case for you too?)

```
ERROR: dsl0 does currently not exist, not even trying to start SQM on nothing.
```
_(to see the logs, logging for sqm-service must be activated (luci > netzowrk > sqm))_

**the interface `dsl0` is not yet present at startup.
To solve this, I have created the following init-script:**

`/etc/init.d/run_on_boot`
```SH
#!/bin/sh /etc/rc.common
# Example script
# Copyright (C) 2007 OpenWrt.org

START=99
STOP=99

start() {
        echo start
        # take a look at https://openwrt.org/docs/techref/initscripts#example_init_script
        # restart sqm 180sec after reboot, to grant working with restarted network-interface

        restart_sqm_service() {
    WAN_IP_EXISTS=0

    while [ "${WAN_IP_EXISTS}" -eq "0" ]; do
        CURRENT_WAN_IP=$(ubus call network.interface.wan_vdsl status | grep -E "\"address\"\:[[:space:]]\"")

        if [ -z "${CURRENT_WAN_IP}" ]; then
            WAN_IP_EXISTS=0
            sleep 60
        else
            WAN_IP_EXISTS=1
            /etc/init.d/sqm restart
        fi
    done
}

restart_sqm_service

}

stop() {
        echo "stop"
        # commands to kill application
}
```
and enable it with `/etc/init.d/run_on_boot enable`

_actually the sqm-service should be able to do this automatically, so my expectation. does anyone know why not ?_

have a happy day everyone

**Comment by Djfe** [comment]
Does your DSL not require a VLAN? (and an according interface name like dsl0.7)

```
ERROR: dsl0 does currently not exist, not even trying to start SQM on nothing.
```

When does your dsl0 interface come up (seconds after booting/first message in logread)? 
When does this message occur in logread in comparison?

For me dsl0 comes up in 2min10s after the first kernel message on my AVM Fritz!Box7520
I don't have the error message above, but I'm using SQM on my dsl0.7 interface

**Comment by gottaeat** [comment]
i have the international version of this model, i had the exact same performance issue.
isp router gets 71/14 down and up, fritzbox gets 45/14.

## what didn't solve it:
1. switching to the proprietary firmwares extracted from a variety of images.
2. setting `dsl0` `qdisc` to `noqueue`

## what fixed it
1. software offloading of routing/NAT
2. `eth0` `qdisc` set to `noqueue`

and bam, exact performance of the ISP router.

**Comment by brada4** [comment]
Please also try 3 different steering settings in luci/network/interfaces/global
Probably observe CPU usage with htop - in general load should be spread among all cpus.
